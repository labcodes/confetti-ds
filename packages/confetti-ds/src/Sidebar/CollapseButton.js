import React from "react";
import PropTypes from "prop-types";

import Icon from "../Icon";

export default class CollapseButton extends React.Component {
  static propTypes = {
    /** Callback action to be executed when the CollapseButton is clicked. */
    onClick: PropTypes.func.isRequired,
  };

  render() {
    const { onClick } = this.props;

    return (
      <button
        type="button"
        onClick={onClick}
        className="lab-narrow-sidebar__collapse"
      >
        <Icon
          type="menu-collapse"
          className="lab-narrow-sidebar__collapse-icon"
        />
      </button>
    );
  }
}
