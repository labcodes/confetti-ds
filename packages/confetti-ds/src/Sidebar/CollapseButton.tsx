import React from "react";
import PropTypes from "prop-types";

import Icon from "../Icon";

/**
 *
 * @param onClick Callback action to be executed when the CollapseButton is clicked.
 * @returns {JSX.Element}
 * @constructor
 *
 */
export default function CollapseButton({ onClick }) {
  return (
    <button
      type="button"
      onClick={onClick}
      className="lab-narrow-sidebar__collapse"
    >
      <Icon
        type="menu-collapse"
        className="lab-narrow-sidebar__collapse-icon"
      />
    </button>
  );
}

CollapseButton.propTypes = {
  /** Callback action to be executed when the CollapseButton is clicked. */
  onClick: PropTypes.func.isRequired,
};
