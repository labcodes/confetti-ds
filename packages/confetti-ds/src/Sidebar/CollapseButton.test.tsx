import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

import CollapseButton from "./CollapseButton";

describe("CollapseButton", () => {
  it("renders with base props", async () => {
    const renderedComponent = renderer
      .create(<CollapseButton onClick={() => {}} />)
      .toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it("calls props.onClick when clicked", async () => {
    const mockOnClick = jest.fn();
    const shallowButton = shallow(<CollapseButton onClick={mockOnClick} />);
    shallowButton.simulate("click");
    expect(mockOnClick.mock.calls.length).toEqual(1);
  });
});
