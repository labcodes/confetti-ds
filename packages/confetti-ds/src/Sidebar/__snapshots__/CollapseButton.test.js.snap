// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapseButton renders with base props 1`] = `
<button
  className="lab-narrow-sidebar__collapse"
  onClick={[Function]}
  type="button"
>
  <span
    className="lab-icon lab-icon--menu-collapse lab-narrow-sidebar__collapse-icon"
  />
</button>
`;
